src/pcapi/core/offers/api.py

```diff
@@ -385,7 +386,7 @@ def delete_stock(stock: Stock) -> None:
     cancelled_bookings = []
     for booking in stock.bookings:
         if not booking.isCancelled and not booking.isUsed:
-            booking.isCancelled = True
+            cancel_bookings_when_deleting_stock(booking)
             cancelled_bookings.append(booking)
 
     repository.save(stock, *cancelled_bookings)
```

Comment:
```suggestion
            cancel_bookings_when_deleting_stock(booking)
            cancelled_bookings.append(booking)

    repository.save(stock)
```

Il n'est plus nécessaire d'appeler `repository.save()` sur les réservations, puisque `cancel_bookings_when_deleting_stock()` s'en charge.